src/aleph/web/controllers/storage.py

```diff
@@ -118,18 +108,53 @@ def content(self) -> Union[str, bytes]:
 
 class MultipartUploadedFile:
     _content: Optional[bytes]
+    metadata: bool
+    size = 0
 
-    def __init__(self, file_field: FileField):
+    def __init__(self, file_field: FileField, metadata: bool):
+        self.metadata = metadata
         self.file_field = file_field
+        self.MAX_SIZE = (
+            MAX_UPLOAD_FILE_SIZE
+            if self.metadata
+            else MAX_UNAUTHENTICATED_UPLOAD_FILE_SIZE
+        )
 
         try:
-            content_length_str = file_field.headers["Content-Length"]
-            self.size = int(content_length_str)
+            if self.metadata:
+                content_length_str = file_field.headers["Content-Length"]
+                self.size = int(content_length_str)
+                self._content = self.file_field.file.read()
```

Comment:
This should use `self.read_file_with_max_size()` and possibly throw an exception, if we were not able to read everything